lib/Setup Thermostats.js

```diff
@@ -104,13 +105,13 @@ exports.setupThermostat = function (that, services)
 				{
 					var updateTemp = (that.platformConfig.temperatureUnits.includes("F"))
 							? Math.round((newHomekitValue * 1.8) + 32)
-							: newHomekitValue					
+							: newHomekitValue
 					
-						if (TargetState.value == 2) // 2 = homekit cool
+						if (this.targetState == 2) // 2 = homekit cool

```

Comment:
I think this ties to my prior comment re: line 66 - here, the "this" is different so "this.targetState" always evaluate to "null" since "this" will be local to the function being called on line 103?

Reply:
Yeah I’ll likely refactor. Attempting to follow best practices as shown [here](https://gist.github.com/oznu/0c32d10473c1c26e6e2aa57809b81895) with `this.isOn` I’m testing it live so I’ll circle back on this one.

Reply:
as discussed, reverted this. Now adding a 50ms delay to ensure any scenes will set mode before temperature.